mqtt/mqtt-broker/src/configuration.rs

```diff
@@ -0,0 +1,304 @@
+use config::{Config, ConfigError, File, FileFormat};
+use lazy_static::lazy_static;
+use regex::Regex;
+use serde::{Deserialize, Deserializer};
+use std::path::Path;
+use std::time::Duration;
+
+pub const DEFAULTS: &str = include_str!("../config/default.json");
+
+#[derive(Debug, Deserialize)]
+pub enum QueueFullAction {
+    #[serde(rename(deserialize = "drop_new"))]
+    DropNew,
+
+    #[serde(rename(deserialize = "drop_old"))]
+    DropOld,
+
+    #[serde(rename(deserialize = "disconnect"))]
+    Disconnect,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct InflightMessages {
+    max_count: u32,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct RetainedMessages {
+    max_count: u32,
+
+    #[serde(with = "humantime_serde")]
+    expiration: Duration,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct SessionMessages {
+    #[serde(deserialize_with = "humansize")]
+    max_message_size: u64,
+
+    max_count: u32,
+
+    #[serde(deserialize_with = "humansize")]
+    max_total_space: u64,
+
+    when_full: QueueFullAction,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct SessionPersistence {
+    file_path: String,
+
+    #[serde(with = "humantime_serde")]
+    time_interval: Duration,
+    unsaved_message_count: u32,
```

Comment:
Somewhere fields are separated with new line, somewhere not. Could you please make it more consistent?